arches/app/tasks.py

```diff
@@ -629,6 +629,108 @@ def reverse_etl_load(loadid):
     module.reverse_load(loadid)
 
 
+@shared_task
+def update_resource_instance_data_based_on_graph_diff(
+    initial_graph, updated_graph, user_id
+):
+    logger = logging.getLogger(__name__)
+    user = User.objects.get(id=user_id)
+
+    try:
+        # delete extra tiles on nodegroup cardinality change
+        initial_nodegroup_ids_to_cardinality = {}
+        for initial_nodegroup in initial_graph["nodegroups"]:
+            initial_nodegroup_ids_to_cardinality[initial_nodegroup["nodegroupid"]] = (
+                initial_nodegroup["cardinality"]
+            )
+
+        nodegroups_with_extra_tiles = []
+        for updated_nodegroup in updated_graph["nodegroups"]:
+            if (
+                updated_nodegroup["cardinality"] == "1"
+                and initial_nodegroup_ids_to_cardinality.get(
+                    updated_nodegroup["nodegroupid"]
+                )
+                == "n"
+            ):
+                nodegroups_with_extra_tiles.append(updated_nodegroup["nodegroupid"])
+
+        for tile in models.TileModel.objects.exclude(sortorder=0).filter(
```

Comment:
I don't think we can depend on the first tile having a sortorder of exactly zero. For a long time the bulk data manager created tiles with a null sortorder. A sortorder sequence could start anywhere other 0, also.